for-research/public-likelihoods/public-likelihoods.ipynb

```diff
@@ -0,0 +1,936 @@
+    "\n",
+    "# we can save this modified workspace like this:\n",
+    "w.SaveAs(\"reparameterized_ws.root\")\n",
+    "\n",
+    "# restore the state of the workspace by replacing the function again with the original poi\n",
+    "# Can get a list of all functions in a workspace using the `factors()` method on the workspace:\n",
+    "w.factors()[\"myFunc\"].Replace(poiName) #if you were to now Print the workspace, `myFunc` will appear in the 'scratch' folder, because it isn't part of a pdf any more"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14fc2625-8e28-481a-a7c1-cdc8c1570706",
+   "metadata": {},
+   "source": [
+    "## 2.2. Creating a new workspace: Selecting a subset of the channels\n",
+    "\n",
+    "We can use the `xRooNode::reduced` method to create a shallow-copy of a multi-channel pdf for a subset of the channels. This new pdf is not added to the workspace (but it can be), but we will show how we can instead add it to a new workspace, along with the obsData associated to the selected channels:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "b19bc20b-36d2-4446-b1a3-a021ee2696b4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 7 channels in reduced model\n",
+      "[#1] INFO:ObjectHandling -- RooWorkspace::import(newws) importing dataset obsData\n",
+      "newws: RooWorkspace::newws\n",
+      "!models\n",
+      " 0) simPdf_reduced : RooSimultaneous::simPdf_reduced\n",
+      "!datasets\n",
+      " 1) obsData : RooDataSet::obsData[channelCat,obs_x_reg_3l_ttZ,obs_x_reg_4l_DF,obs_x_reg_4l_FF_CR_e_hf,obs_x_reg_4l_FF_CR_e_other,obs_x_reg_4l_FF_CR_mu_hf,obs_x_reg_4l_SF,obs_x_reg_4l_ZZ,weight:weight] = 28 entries (3430 weighted)\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Info in <xRooNode::SaveAs>: newws saved to reduced_ws.root\n"
+     ]
+    }
+   ],
+   "source": [
+    "newModel = w[\"models/simPdf\"].reduced(\"reg_3l_ttZ,reg_4l_*\") # specify subset of channels as comma separated list, note you can use wildcards.\n",
+    "print(\"There are\",newModel.bins().size(),\"channels in reduced model\")\n",
+    "\n",
+    "# we can save this reduced pdf to a workspace of its own, if we wish ....\n",
+    "w2 = ROOT.xRooNode(\"RooWorkspace\",\"newws\",\"my new workspace\") # Creates a new RooWorkspace (name=newws,title=my new workspace), wrapped by an xRooNode\n",
+    "w2.Add(newModel) # note: workspaces create an owned copy of things that get added to them\n",
+    "w2.Add(newModel.datasets()[\"obsData\"]) # also add the subset of the obsData dataset that is relevant to newModel\n",
+    "w2.Print() # inspect the content of this workspace before saving\n",
+    "w2.SaveAs(\"reduced_ws.root\") # BUG: json serializer currently crashing, so can only save as a ROOT file at the moment"
```

Comment:
Now I understand why we used ROOT I/O earlier :)  In what version of ROOT was this a problem?

Reply:
ROOT master. Carsten I think is fixing it.